ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -980,37 +991,30 @@ static int ompi_osc_rdma_query_btls (ompi_communicator_t *comm, ompi_osc_rdma_mo
        in general usage. */
     btls_to_use = opal_argv_split (ompi_osc_rdma_full_connectivity_btls, ',');
     if (btls_to_use) {
-        /* rdma and atomics are only supported with BTLs at the moment
-         * If a btl does not support remote completion, it cannot be used as the primary btl.
-         * It can still be selected as an alternate btl */
+        mca_btl_base_selected_module_t *item;
+
+        selected_btl = NULL;
+
+        /* rdma and atomics are only supported with BTLs at the moment */
         OPAL_LIST_FOREACH(item, &mca_btl_base_modules_initialized, mca_btl_base_selected_module_t) {
             for (int i = 0 ; btls_to_use[i] ; ++i) {
                 if (0 != strcmp (btls_to_use[i], item->btl_module->btl_component->btl_version.mca_component_name)) {
                     continue;
                 }
 
-                if ((item->btl_module->btl_flags & (MCA_BTL_FLAGS_RDMA)) == MCA_BTL_FLAGS_RDMA &&
-                    (item->btl_module->btl_flags & (MCA_BTL_FLAGS_ATOMIC_FOPS | MCA_BTL_FLAGS_ATOMIC_OPS | MCA_BTL_FLAGS_RDMA_REMOTE_COMPLETION))) {
-                    if (!selected_btl || item->btl_module->btl_latency < selected_btl->btl_latency) {
+                if (ompi_osc_rdma_check_accelerated_btl(item->btl_module)) {
+                    if (NULL == selected_btl || item->btl_module->btl_latency < selected_btl->btl_latency) {
                         selected_btl = item->btl_module;
                     }
                 }
             }
         }
 
         opal_argv_free (btls_to_use);
-    }
 
-    if (NULL != selected_btl) {
-        if (module) {
-            ompi_osc_rdma_selected_btl_insert(module, selected_btl, 0);
-            module->btls_in_use = 1;
-            module->use_memory_registration = selected_btl->btl_register_mem != NULL;
```

Comment:
It seems the setting of `module->use_memory_registration` was removed. Is that OK?

Reply:
To try and reduce code duplication, this block of code is all in the code after the `btl_selected_completion` label.